src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -397,20 +408,16 @@ export class SavedObjectsRepository {
     fields,
     namespace,
     type,
+    filter,
   }: SavedObjectsFindOptions): Promise<SavedObjectsFindResponse<T>> {
     if (!type) {
       throw new TypeError(`options.type must be a string or an array of strings`);
     }
 
     const types = Array.isArray(type) ? type : [type];
-    const allowedTypes = types.filter(t => this._allowedTypes.includes(t));
-    if (allowedTypes.length === 0) {
-      return {
-        page,
-        per_page: perPage,
-        total: 0,
-        saved_objects: [],
-      };
+    const notAllowedTypes = types.filter(t => !this._allowedTypes.includes(t));
+    if (notAllowedTypes.length > 0) {
+      throw new TypeError(`options.type ${notAllowedTypes.join(', ')} are not allowed`);
```

Comment:
Please use `SavedObjectsErrorHelpers.createUnsupportedTypeError(type);` to construct the error.